ceph/ceph/upgrade.sh

```diff
@@ -0,0 +1,57 @@
+#!/bin/bash
+
+NAMESPACE=${NAMESPACE:-"ceph"}
+TIMEOUT=${TIMEOUT:-10}
+UPGRADE_ORDER="mgr mon osd mds rgw rbd-provisioner"
+KUBECTL="kubectl"
+
+function wait_for_pod_ready {
+    pod=$1
+    i=${TIMEOUT}
+    echo "Wait for Pod "$pod" to be ready"
+    while [ $i -gt 0 ]; do
+        ready=$(kubectl get pod ${pod} --namespace=${NAMESPACE} -o template --template="{{ (index .status.containerStatuses 0).ready }}")
+        if [ $? -ne 0 ]; then 
+            # bad pod
+            sleep 1
+            i=$(( $i-1 ))
+            continue
+        fi
+
+        if [ ${ready} == "true" ]; then 
+            break
+        fi
+        sleep 1
+        i=$(( $i-1 ))
+    done
+}
+
+function wait_for_new_pod {
+    new_pods=$(kubectl get pod --namespace=${NAMESPACE} -l application=ceph -l component=${component} -o template --template="{{range .items}} {{.metadata.name }} {{end}}")
+    for n in ${new_pods}; do
+        found=0
+        for p in ${pods}; do
+            if [ ${n} == ${p} ]; then
+                found=1
+                break
+            fi
+        done
+        if [ ${found} -eq 0 ]; then
+           wait_for_pod_ready ${n}
+        fi
+    done
+}
+
+function main {
+    for component in ${UPGRADE_ORDER}; do
+        echo "Upgrading " ${component}
```

Comment:
Even though this works, why not including `${component}` inside the double quotes? The construction looks weird.